test/public/test/data-remote.js

```diff
@@ -155,6 +155,31 @@ asyncTest('submitting form with data-remote attribute', 4, function() {
     .trigger('submit');
 });
 
+asyncTest('submitting form with data-remote attribute should include inputs in a fieldset only once', 3, function() {
+  $('form[data-remote]')
+    .append('<fieldset><input name="items[]" value="Item" /></fieldset>')
+    .bind('ajax:success', function(e, data, status, xhr) {
+      App.assertCallbackInvoked('ajax:success');
+      equal(data.params.items.length, 1, 'ajax arguments should only have the item once')
+      App.assertPostRequest(data);
+    })
+    .bind('ajax:complete', function() { start() })
+    .trigger('submit');
+});
+
+asyncTest('form fields from disabled fieldsets will not be included', 4, function() {
+  $('form[data-remote]')
+    .append('<fieldset disabled="disabled"><input name="not_here" value="wont_show_up" /></fieldset>')
```

Comment:
We need to put a required input here to make sure that it doesn't prevent the form submission